_posts/2016-04-05-extension-framework-game-plan.md

```diff
@@ -0,0 +1,154 @@
+---
+layout: post
+cover: 'assets/images/general-cover-3.jpg'
+title: Extension Framework Game Plan
+tags: docs
+subclass: 'post tag-docs'
+categories: 'elixir'
+author: 'Pikender'
+navigation: true
+logo: 'assets/images/nectar-cart.png'
+---
+
+>
+The post belongs to _NectarCommerce and Extension Framework Awareness_ Series
+>
+1. _[NectarCommerce Vision](http://vinsol.com/blog/2016/04/08/nectarcommerce-vision/)_
+1. **Extension Framework Game Plan**
+1. Introduction to Writing Macros
+1. Running Multiple Phoenix Apps Together
+1. Ecto Model Extension
+1. Phoenix Router Extension
+1. Phoenix View Extension
+1. Extension Approach Explained
+1. Developer Experience and Workflow developing Favorite Product Extension
+1. Developer Experience and Workflow testing Favorite Product Extension
+
+## What will be NectarCommerce
+
+>
+Off-the-shelf Opensource E-commerce application for building online store.
+>
+Provides an Extension Framework to support features not included in core as extensions.
+>
+Strives for un-obstrusive parallel development of NectarCommerce and Extensions
+
+NectarCommerce is committed to provide a ready-to-use e-commerce solution but definition of 100% is different under different business domains. It aims to solve common use-cases as part of the project and relying on extension framework to tap the rest.
+
+## Extension Framework Game Plan
+
+Let's validate and list the capabilities needed in Extension Framework through an **extension** which provides feature **to favorite products by Users.**
+
+_**Favorite Product Extension Requirements:**_
+
+- **Ability to mark and unmark a Product as favorite**
+  - a View _probably_ showing all products with Ability to mark/unmark product as favorite
+  - a controller action preparing the view
+  - a route exposing the controller / view through Web
+  - a controller action handling mark product as favorite
+  - a route exposing the controller / view through Web
+  - a controller action handling unmark already favorited product
```

Comment:
a controller action handling unmarking of a favorite product (removing it from the list of favorites)